crates/storage/src/connection.rs

```diff
@@ -265,6 +265,15 @@ impl<'inner> Transaction<'inner> {
         state_update::contract_class_hash(self, block_id, contract_address)
     }
 
+    pub fn insert_contract_class_hash(
+        &self,
+        block_number: BlockNumber,
+        contract_address: ContractAddress,
+        class_hash: ClassHash,
+    ) -> anyhow::Result<()> {
+        state_update::insert_contract_class_hash(self, block_number, contract_address, class_hash)
+    }
+
```

Comment:
Could you replace this by using the `insert_state_diff` instead? I realize this is more convenient since this is all you require for the tests, but I think keeping the storage write API restricted to what is required by sync will be beneficial in the long run.

Since there are often hidden requirements for the database storage (like foreign key relations etc), which are also subject to change, having these extra methods will be more brittle. It will also help force us to create more realistic test data.

Something like the following should work for the tests:
```rust
let state_diff = StateDiff::default()
    .add_deployed_contract(contract_address, class_hash);
tx.insert_state_diff(block_number, state_diff).unwrap();
```

Reply:
My bad, totally missed that I can do the same with existing method.

Reply:
No problem; should probably be documented more explicitly somewhere since this is a recent "idea" :)

Reply:
Done.